darts/models/theta.py

```diff
@@ -115,14 +120,241 @@ def predict(self, n: int) -> 'TimeSeries':
+            _, self.seasonality = extract_trend_and_seasonality(new_ts, self.season_period,
+                                                                model=self.season_mode.value)
+            new_ts = remove_seasonality(new_ts, self.season_period, model=self.season_mode.value)
+
+        ts_values = new_ts.univariate_values()
+        if (ts_values <= 0).any():
+            self.model_mode = Model.ADDITIVE
+            self.trend_mode = Trend.LINEAR
+            logger.warn("Time series has negative values. Fallback to additive and linear model")
+
+        # Drift part of the decomposition
+        if self.trend_mode is Trend.LINEAR:
+            linreg = ts_values
+        else:
+            linreg = np.log(ts_values)
+        self.drift = np.poly1d(np.polyfit(np.arange(self.length), linreg, 1))
+        theta0_in = self.drift(np.arange(self.length))
+        if self.trend_mode is Trend.EXPONENTIAL:
+            theta0_in = np.exp(theta0_in)
+
+        if (theta0_in > 0).all() and self.model_mode is Model.MULTIPLICATIVE:
+            theta_t = (ts_values ** self.theta) * (theta0_in ** (1 - self.theta))
+        else:
+            if self.model_mode is Model.MULTIPLICATIVE:
+                logger.warn("Negative Theta line. Fallback to additive model")
+                self.model_mode = Model.ADDITIVE
+            theta_t = self.theta * ts_values + (1 - self.theta) * theta0_in
+
+        # SES part of the decomposition.
+        self.model = hw.SimpleExpSmoothing(theta_t).fit()
+        theta2_in = self.model.fittedvalues
+
+        if (theta2_in > 0).all() and self.model_mode is Model.MULTIPLICATIVE:
+            self.fitted_values = theta2_in**self.wses * theta0_in**self.wdrift
+        else:
+            if self.model_mode is Model.MULTIPLICATIVE:
+                self.model_mode = Model.ADDITIVE
+                logger.warn("Negative Theta line. Fallback to additive model")
+                theta_t = self.theta * new_ts.values() + (1 - self.theta) * theta0_in
+                self.model = hw.SimpleExpSmoothing(theta_t).fit()
+                theta2_in = self.model.fittedvalues
+            self.fitted_values = self.wses * theta2_in + self.wdrift * theta0_in
+        if self.is_seasonal:
+            if self.season_mode is Season.ADDITIVE:
+                self.fitted_values += self.seasonality.univariate_values()
+            elif self.season_mode is Season.MULTIPLICATIVE:
+                self.fitted_values *= self.seasonality.univariate_values()
+        # Fitted values are the results of the fit of the model on the train series. A good fit of the model
+        # will lead to fitted_values similar to ts. But one cannot see if it overfits.
+        if self.normalization:
+            self.fitted_values *= self.mean
+        # Takes too much time to create a TimeSeries
+        # Overhead: 30% ± 10 (2-10 ms in average)
+        self.fitted_values = TimeSeries.from_times_and_values(ts.time_index(), self.fitted_values)
```

Comment:
You could build the `TimeSeries` only in the backtesting method, this way the time overhead is paid only for backtesting and not for simply fitting the model.